packages/xrpl/src/sugar/submit.ts

```diff
@@ -116,32 +122,53 @@ async function submitRequest(
  * validated ledger, or the transaction's lastLedgerSequence has been surpassed by the
  * latest ledger sequence (meaning it will never be included in a validated ledger).
  */
+// eslint-disable-next-line max-params, max-lines-per-function -- fn needs to display and do with more information.
 async function waitForFinalTransactionOutcome(
   client: Client,
   txHash: string,
+  lastLedger: number,
+  submissionResult: string,
 ): Promise<TxResponse> {
   await sleep(LEDGER_CLOSE_TIME)
 
-  const txResponse = await client.request({
-    command: 'tx',
-    transaction: txHash,
-  })
-  if (txResponse.result.validated) {
-    return txResponse
-  }
+  const latestLedger = await client.getLedgerIndex()
 
-  const txLastLedger = txResponse.result.LastLedgerSequence
-  if (txLastLedger == null) {
-    throw new XrplError('LastLedgerSequence cannot be null')
+  if (lastLedger < latestLedger) {
+    throw new XrplError(
+      `The latest ledger sequence ${latestLedger} is greater than the transaction's LastLedgerSequence (${lastLedger}).
+      Preliminary result: ${submissionResult}`,
```

Comment:
This should use string concatenation so you don't have the `\n` in the error message